### ISSUE TITLE
react/prop-types doesn't work correctly with Flow

### ISSUE THREAD

**Issue by pbondoer** [yannickcr/eslint-plugin-react#2333]
Example code:

```js
// @flow

type Role = 'owner' | 'admin' | 'user';

type Props = {|
  contacts: { [role: Role]: string },
|};

export default class MyClass extends React.Component<Props> {
  render() {
    return (
      <>
        {contacts.owner}
      </>
    );
  }
}
```

Error:

```
error  'contacts.owner' is missing in props validation  react/prop-types
```

Using `eslint-plugin-react@7.14.2` and `eslint@5.16.0`.

This does not happen with `eslint-plugin-react@7.13.0`.

**Comment by yannickcr** [comment]
Seems to buplicate of https://github.com/yannickcr/eslint-plugin-react/issues/2330.

It should be fixed in the next release.
